source/common/common/utility.h

```diff
@@ -18,6 +18,15 @@
 #include "absl/strings/string_view.h"
 
 namespace Envoy {
+
+/**
+ * Retrieve string description of error code, wraps strerror on POSIX systems and FormatMessage on
+ * Windows
+ * @param int error code
+ * @return const std::string error detail description
+ */
+const std::string errorDetails(int error_code);
```

Comment:
Open to moving this somewhere else (its own error utility lib? or otherwise) or renaming